corelib/src/Features2d.cpp

```diff
@@ -2615,6 +2619,91 @@ cv::Mat SuperPointTorch::generateDescriptorsImpl(const cv::Mat & image, std::vec
 }
 
 
+//////////////////////////
+//SuperPointRpautrat
+//////////////////////////
+SuperPointRpautrat::SuperPointRpautrat(const ParametersMap & parameters) :
+		superpointDir_(Parameters::defaultSuperPointRpautratDir()),
+		threshold_(Parameters::defaultSuperPointRpautratThreshold()),
+		nms_(Parameters::defaultSuperPointRpautratNMS()),
+		minDistance_(Parameters::defaultSuperPointRpautratNMSRadius()),
+		cuda_(Parameters::defaultSuperPointRpautratCuda())
+{
+	parseParameters(parameters);
+}
+
+SuperPointRpautrat::~SuperPointRpautrat()
+{
+}
+
+void SuperPointRpautrat::parseParameters(const ParametersMap & parameters)
+{
+	Feature2D::parseParameters(parameters);
+
+	std::string previousPath = superpointDir_;
+#ifdef RTABMAP_TORCH
+	bool previousCuda = cuda_;
+#endif
+	
+	Parameters::parse(parameters, Parameters::kSuperPointRpautratDir(), superpointDir_);
+	Parameters::parse(parameters, Parameters::kSuperPointRpautratThreshold(), threshold_);
+	Parameters::parse(parameters, Parameters::kSuperPointRpautratNMS(), nms_);
+	Parameters::parse(parameters, Parameters::kSuperPointRpautratNMSRadius(), minDistance_);
+	Parameters::parse(parameters, Parameters::kSuperPointRpautratCuda(), cuda_);
+	
+
+#ifdef RTABMAP_TORCH
+	if(superPoint_.get() == 0 || superpointDir_.compare(previousPath) != 0 || previousCuda != cuda_)
+	{
+		superPoint_ = cv::Ptr<SPDetectorRpautrat>(new SPDetectorRpautrat(superpointDir_, threshold_, nms_, minDistance_, cuda_));
+	}
+	else
+	{
+		superPoint_->setSuperpointDir(superpointDir_);
+		superPoint_->setThreshold(threshold_);
+		superPoint_->setNMS(nms_);
+		superPoint_->setMinDistance(minDistance_);
+	}

```

Comment:
You may remove these setters because they are not really supported. The model is initialized on the first frame, so any changes here won't matter, the whole model should be re-generated again to use of the new parameters.

It seems that if any parameter changes, we need to delete the previous detector to force it re-initialize on the next frame.